### PR TITLE
test(panel-admin): cover AssignRoleAction — issue #154

### LINKED ISSUE #154: [Testes][panel-admin] Cobrir gaps — 13 itens (1 crítico, 5 altos)
## Contexto

Painel administrativo global (SuperAdmin/Admin). Cobertura atual: 4 testes Feature (CRUD Company, ViewAppointment, EditUserProfile). Ampla superfície Filament sem teste: filtros reativos da `AppointmentResource`, Resources de Consultant/Plan/Price/ContractualPlan, widgets `Metrics/*`, action `AssignRoleAction`, listeners de notificação para admins.

## Objetivo

Fechar gaps identificados na auditoria (`docs/testing/coverage-gap-report.md#panel-admin`).

**Fora de escopo**: browser, DTOs, Enums sem lógica, código já coberto por PHPStan level max / Pint / Rector.

### 🔴 Crítico
- [ ] `app-modules/panel-admin/src/Filament/Resources/Permissions/Actions/AssignRoleAction.php:14` — valida duplicidade, `visible()` só para SuperAdmin; sem teste — Feature — P

### 🟠 Alto
- [ ] `app-modules/panel-admin/src/Filament/Resources/Appointments/Tables/AppointmentsTable.php` — filtros `live(debounce: 500)` por user/consultant/status/company_id/date_range — testar combinações — Feature — M
- [ ] `app-modules/panel-admin/src/Filament/Resources/Appointments/Schemas/AppointmentForm.php` — `live()` + `afterStateUpdated` em `appointment_at` recarrega consultants — Feature — M
- [ ] `app-modules/panel-admin/src/Filament/Resources/Consultants/ConsultantResource.php` — Tabs `hidden(fn ($operation))`, SpatieMediaLibraryFileUpload — sem nenhum teste do Resource — Feature — M
- [ ] `app-modules/panel-admin/src/Filament/Widgets/AppointmentsStatsOverview.php:23,28` — `syncFilters()` via `#[On('appointments-table-filters-changed')]` + `getStats()` com `selectRaw count() filter` — Feature — M
- [ ] `app-modules/panel-admin/src/Listeners/NotifyAdminsOfAppointmentBookedListener.php:17` (e 3 irmãos) — validar destinatários e idempotência; hoje coberto parcialmente por `tests/Feature/Listeners/AdminNotificationsTest.php` — Feature — M

### 🟡 Médio
- [ ] `app-modules/panel-admin/src/Filament/Resources/Companies/Schemas/CompanyForm.php` — slug gerado reativo (`live(onBlur,debounce:500)`) — Feature — P
- [ ] `app-modules/panel-admin/src/Filament/Resources/ContractualPlans/*` — CRUD sem teste — Feature — P
- [ ] `app-modules/panel-admin/src/Filament/Resources/Plans/*` — CRUD sem teste (há teste em `billing` para o mesmo Resource — confirmar se duplicata) — Feature — P
- [ ] `app-modules/panel-admin/src/Filament/Widgets/Metrics/*` (5 widgets) — queries agregadas sem teste — Feature — M
- [ ] `app-modules/panel-admin/src/Actions/GetAdminUsersAction.php` — retorna só Admin/SuperAdmin — Unit — P
- [ ] `app-modules/panel-admin/src/Policies/BetterMailPolicy.php` / `InboundWebhookPolicy.php` — Feature — P

### 🟢 Baixo
- [ ] `app-modules/panel-admin/src/Filament/Widgets/QuickActions.php:21` — `getViewData()` — Feature — P

## Fluxos de usuário impactados

- **Admin global** — gestão de planos, roles, consultants.
- **Notificações admin** — eventos Appointment e UserRegistered.

## Pontos críticos específicos

- [ ] Authorização: ações restritas a SuperAdmin/Admin
- [ ] Filtros reativos não quebram com combinações diferentes
- [ ] Listeners enviam notificação apenas para admins e não duplicam

## Critério de aceite

- Todos os itens 🔴 e 🟠 concluídos.
- `make test` verde.
- `make phpstan` sem regressão.
- `make check` verde no CI.

## Referências

- Relatório: `docs/testing/coverage-gap-report.md#panel-admin`

### PR DESCRIPTION
**Pull request by Clintonrocha98**
## What was tested

- The **Assign Role** action in the admin panel user list:
  - Is visible only to SuperAdmin users (hidden from Admins and other roles)
  - Successfully assigns a chosen role to a target user
  - Shows a conflict notification when the target user already has the selected role, without duplicating the assignment

## Why it matters

The role assignment action is a critical administrative operation. These tests guarantee that only authorized users can trigger it, that it correctly changes user permissions, and that it handles edge cases (duplicate role assignment) gracefully without silent failures.

Closes #154

## Summary by CodeRabbit

* **Tests**
  * Added test coverage for appointment creation form behavior, ensuring consultant selection clears when appointment date changes
  * Added tests for appointment list filtering by user name (partial and empty searches)
  * Added tests for consultant resource pages and form fields (create/edit)
  * Added tests for role-assignment actions visibility and behavior
  * Added tests verifying admin notifications are created when appointments are booked

### CODE REVIEW

File: app-modules/panel-admin/tests/Feature/Filament/Resources/Consultants/ConsultantResourceTest.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+it('renders the consultants list page', function (): void {
+    livewire(ListConsultants::class)
+        ->assertOk();
+});
+
+it('hides the preview tab on the create form', function (): void {
+    livewire(CreateConsultant::class)
+        ->assertDontSee('preview::tab', escape: false);
+});
```
[the remark this answers is not in the document]
Comment (Clintonrocha98) [reply app-modules/panel-admin/tests/Feature/Filament/Resources/Consultants/ConsultantResourceTest.php:24]: Won't fix: `assertDontSee('preview::tab', escape: false)` verifica o atributo `wire:key` que o Filament renderiza no DOM para o tab hidden. Quando `hidden()` retorna `true`, o atributo não aparece no HTML e o teste detecta corretamente a regressão. Checar o label visível "Preview" seria equivalente mas menos preciso, pois o label poderia estar presente em outro contexto da página.

File: app-modules/panel-admin/tests/Feature/Listeners/NotifyAdminsOfAppointmentBookedListenerTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    $this->employee = User::factory()->create();
+    $this->employee->assignRole('employee');
+});
+
+it('sends a database notification to each admin when an appointment is booked', function (): void {
+    $appointment = Appointment::factory()->create(['user_id' => $this->employee->id]);
+
+    $before1 = DB::table('notifications')->where('notifiable_id', $this->admin1->id)->count();
+    $before2 = DB::table('notifications')->where('notifiable_id', $this->admin2->id)->count();
+
```
[the remark this answers is not in the document]
Comment (Clintonrocha98) [reply app-modules/panel-admin/tests/Feature/Listeners/NotifyAdminsOfAppointmentBookedListenerTest.php:25]: Won't fix: os testes usam `RefreshDatabase`, então o banco começa limpo a cada execução — risco de colisão com outros modelos morphable é praticamente nulo. Adicionar `notifiable_type` seria correto formalmente, mas adiciona ruído sem ganho real neste contexto.

**Review by thalesmengue — APPROVED** [review]
